product-web-app/service/roleService.ts

```diff
@@ -0,0 +1,25 @@
+import axios from 'axios';
+import Role from '../model/role.js';
+import logger from './logger.js';
+
+export default class RoleService {
+  async getRoles(): Promise<Role[]> {
+    try {
+      const response = await axios.get('http://localhost:8080/api/job-roles');
+      return response.data;
+    } catch (e) {
+      throw new Error('Failed to get roles from the database');
+    }
+  }
+
+  async findRoleById(id: number): Promise<Role> {
+    try {
+      const response = await axios.get(`http://localhost:8080/api/job-roles/${id}`);
```

Comment:
Remove localhost from base url we already set it in 
https://github.com/dsuszek/developerzy-z-barakow/blob/main/product-web-app/app.ts#L34
 
```suggestion
      const response = await axios.get('/api/job-roles');
      return response.data;
    } catch (e) {
      throw new Error('Failed to get roles from the database');
    }
  }

  async findRoleById(id: number): Promise<Role> {
    try {
      const response = await axios.get(`/api/job-roles/${id}`);
```

Reply:
Addressed.